File: algorithm/src/main/java/baekjoon/p01316/Main.java
package baekjoon.p01316;

import java.util.HashSet;
import java.util.Scanner;

public class Main {

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        String[] inputs = new String[n];

        for (int i = 0; i < inputs.length; i++) {
            inputs[i] = sc.next();
        }

        System.out.println(solve(inputs));
    }

    private static int solve(String[] inputs) {
        int ret = 0;

        for (String input : inputs) {
            if (check(input)) {
                ret++;
            }
        }

        return ret;
    }

    private static boolean check(String str) {
        HashSet<Character> hashSet = new HashSet<>();
        hashSet.add(str.charAt(0));

        for (int i = 1; i < str.length(); i++) {
            char pre = str.charAt(i - 1);
            char current = str.charAt(i);

            if (hashSet.contains(current) && pre != current) {
                return false;
            } else {
                hashSet.add(current);
            }
        }

        return true;
    }

}

File: algorithm/src/main/java/baekjoon/p01316/Main.kt
package baekjoon.p01316

fun main(args: Array<String>) {
    val n = readLine()!!.toInt()
    val inputs = Array(n) { readLine()!! }
    println(solve(inputs))
}

private fun solve(inputs: Array<String>): Int {
    return inputs.count { check(it) }
}

private fun check(str: String): Boolean {
    val hashSet = HashSet<Char>()
    hashSet.add(str[0])

    for (i in 1..str.lastIndex) {
        val pre = str[i - 1]
        val current = str[i]

        if (hashSet.contains(current) && pre != current) {
            return false
        } else {
            hashSet.add(current)
        }
    }

    return true
}